Neatenup RemoveBins and add tests

NickDraper opened the pull request:
1. Added a missing test for RangeUnits
1. Consolidated cross parameter validation
1. Fixed a broken check for when units are the same
1. Only require the ws to have a unit when RangeUnits is set
1. Removed some units from the RangeUnits list as they made no sense

*No changes to how the guts of how removebins works!*

**To test:**

1. Test the validation of RemoveBins
 1. Xmax < Xmin
 1. WorkspaceIndex out of range
 1. Removals using RangeBins
 1. Removals using a workspace without units (use CreateWorkspace) with RangeUnits set to AsInput
 1. RangeUnits no longer contains the following entries ("Empty", "Label", "Time", "Degrees")

Re #18963.

### RELEASE NOTES

*Does not need to be in the release notes.*


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

DavidFair commented:
I've found some issues with the validation - 

- Accepts negative numbers for `XMin` and `XMax`
- Using a sample workspace without units (`AsInput`) and all other settings default. Then setting the RangeUnit to something like `dSpacing` gives an incorrect error message about converting from TOF
- If you set `XMax` to a high value (say 999999) the log incorrectly states `XMin` is greater than the largest value

DavidFair commented:
@NickDraper Is this for 3.10 release or the next maintenance cycle?

NickDraper commented:
The improvements here have not changed the previous validation, just added missing tests, and moved the previous validation code to the standard validateInputs location, so it can be checked before exec.

Moved to next release as this is only a minor benefit, and not worth any risk at this point in the release.
